releases/release-5.0.4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: TiDB 5.0.4 Release Notes
+---
+
+# TiDB 5.0.4 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 5.0.4
+
+## Compatibility Changes
+
++ TiDB
+
+    - Revert #19341 to avoid #24326 [#26258](https://github.com/pingcap/tidb/pull/26258)
+
+## Improvements
+
++ TiDB
+
+    - Trigger auto-analyze based on histogram row count [#26707](https://github.com/pingcap/tidb/pull/26707)
+
++ TiKV
+
+    - separate read write ready to reduce read latency [#10620](https://github.com/tikv/tikv/pull/10620)
+
+## Bug Fixes
+
++ TiDB
+
+    - tidb panic while query hash partition table with is null condition [#26963](https://github.com/pingcap/tidb/pull/26963)
+    - planner: fix the unstable test case TestAnalyzeIncremental [#26851](https://github.com/pingcap/tidb/pull/26851)
+    - Keep the overflow check logic same with mysql [#26724](https://github.com/pingcap/tidb/pull/26724)
+    - Fix wrong charset and collation for case when function [#26672](https://github.com/pingcap/tidb/pull/26672)
+    - Fix the case that TiDB would return the wrong result when the children of the UNION contain pure NULL values [#26571](https://github.com/pingcap/tidb/pull/26571)
+    - Fix the issue that greatest(datetime) union null returns empty string [#26565](https://github.com/pingcap/tidb/pull/26565)
+    - fix incompatible last_day func behavior in sql mode [#26000](https://github.com/pingcap/tidb/pull/26000)
+    - Fix the issue that committing pessimistic transactions may report write-conflict errors. [#25974](https://github.com/pingcap/tidb/pull/25974)
+    - Fix the wrong result after "insert ignore on duplicate update" in the secondary index has the same column in primary key but has prefix length [#25905](https://github.com/pingcap/tidb/pull/25905)
+    - Fix the wrong result for "insert ignore duplicate up" on partition table when handle changed and duplicated on secondary index [#25859](https://github.com/pingcap/tidb/pull/25859)
+    - fix wrong enum key in point get [#24772](https://github.com/pingcap/tidb/pull/24772)
+    - fix dml_batch_size doesn't load the global variable [#24731](https://github.com/pingcap/tidb/pull/24731)
+    - planner: filter conflict read_from_storage hints [#24374](https://github.com/pingcap/tidb/pull/24374)
+    - fix wrong type infer for agg function when type is null. [#24354](https://github.com/pingcap/tidb/pull/24354)
+    - fix wrong flen infer for bit constant [#24267](https://github.com/pingcap/tidb/pull/24267)
+    - fix some potential panic in statistics [#24061](https://github.com/pingcap/tidb/pull/24061)
+    - fix approx_percent panic on bit column [#23703](https://github.com/pingcap/tidb/pull/23703)
+    - fix get var expr when session var is hex literal [#23373](https://github.com/pingcap/tidb/pull/23373)
+    - fix unexpected constant fold when year compare string. [#23336](https://github.com/pingcap/tidb/pull/23336)
+
```

Comment:
+ TiKV
- RaftStore Snapshot GC fix: fix the issue that snapshot GC missed GC snapshot files when there's one snapshot file failed to be GC-ed. [#10872](https://github.com/tikv/tikv/pull/10872)